### PR TITLE
[client/catapult]: AddResolvedAddresses is not properly applying resolutions from previous transactions

### PR DESCRIPTION
**Pull request by Jaguar0625**
problem: if a resolution entry is added for a transaction, it should stay in scope for the rest of the block
              AddResolvedAddresses is incorrectly scoping a resolution entry to the transaction that triggered it
    solution: extend the scope of a resolution entry to last until the end of the block

### CODE REVIEW

File: client/catapult/extensions/addressextraction/src/AddressExtractor.cpp [hunk 1]
```diff
@@ -55,17 +55,38 @@ namespace catapult { namespace addressextraction {
...
+					// there is not enough information to determine whether or not 1 has occured:
+					//     1. old alias is used
+					//     2. alias is unlinked
+					//     3. new alias is linked
+					//     4. new alias is used
+					// in this ambiguous situation, include the old alias (potential false positive)
+
+					// for a non-aggregate transaction, a single unresolved address can never map multiple resolved addresses
+					if (!resolutionEntry.Source.SecondaryId)
+						shouldAddPreviousResolution = false;
```
Comment (gimre-xymcity) [inline client/catapult/extensions/addressextraction/src/AddressExtractor.cpp:80]: hmm, why is this needed?

Comment (Jaguar0625) [reply client/catapult/extensions/addressextraction/src/AddressExtractor.cpp:80]: in the case it is a non-aggregate transaction that is using an alias, either:
1. the transaction will be first one using the alias and have a resolution entry with matching primaryId
2. the transaction will NOT be first one using the alias (previous transaction in block would have used it and created resolution entry), so it should use previous resolution entry for address

in case of 1, previous resolution entry isn't needed, so shouldAddPreviousResolution is set to false

**Review by gimre-xymcity — APPROVED** [review]
